### PR TITLE
Fix Dockerfile task up-to-date logic

### PR DESCRIPTION
**Pull request by orzeh**
Should fix issues reported in #357

### CODE REVIEW

**Comment by cdancy** [comment]
Let me see if I understand the logic here correctly: we are collecting all the instructions as an input property and so if this should change further downstream, meaning the instructions, this property will become out-of-date and re-trigger the build?

**Comment by orzeh** [comment]
@cdancy yes. See the test case.

**Comment by cdancy** [comment]
LGTM. Merging now.

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Dockerfile extends DefaultTask {
     File destFile = project.file("$project.buildDir/docker/Dockerfile")
 
     Dockerfile() {
-
+        inputs.property("instructionsCache", { instructions.collect { it.build() }.join('\n') })
```
Comment (childnode) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy]: time passes by ... so my comment https://github.com/bmuschko/gradle-docker-plugin/issues/357#issuecomment-321316089 is now outdated, but perhaps still valid?!

**Pull request merged**